BinaryTreeNodeTest.java

```diff
@@ -0,0 +1,76 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BinaryTreeNodeTest {
+    private BinaryTreeNode<Integer> n1, n2, n3, n4, n5, n6, n7, n8;
+
+    @BeforeEach
+    void setUp() {
+        // Set up the tree
+        n1 = new BinaryTreeNode<>(1, null, null);
+        n2 = new BinaryTreeNode<>(5, null, null);
+        n3 = new BinaryTreeNode<>(3, n1, n2);
+        n4 = new BinaryTreeNode<>(14, null, null);
+        n5 = new BinaryTreeNode<>(9, n3, n4);
+        n6 = new BinaryTreeNode<>(19, null, null);
+        n7 = new BinaryTreeNode<>(18, null, n6);
+        n8 = new BinaryTreeNode<>(16, n5, n7);
+    }
+
+    @Test
+    void findAncestorsTest() {
+        // Normal case
+        assertEquals(n8.findAncestors(18), Collections.singletonList(16));
```

Comment:
If you put a lot of assertions in a single test method you will end up with errors equivalent to "something's wrong with findAncestors", after which you will have to go through the message in more details to find which one failed.  Plus if the first assertion fails, the other ones will not be executed at all, so you will have to rerun them after fixing first one to find out if something else fails.

Each assertion should be in a separate test method with name describing the condition checked. This will give you trivially understandable error messages like "test such and such failed", immediately telling you which case went wrong. Also with "one assertion per method" approach all tests will be executed and you will get status of all test cases even if some of them fail.

Could you please split the tests to something like `findAncestors_normalCase`, `findAncestors_leaf` etc?